pkg/universe.dagger.io/examples/todoapp/plan.cue

```diff
@@ -0,0 +1,49 @@
+package todoapp
+
+import (
+	"dagger.io/dagger"
+	"universe.dagger.io/netlify"
+	"universe.dagger.io/git"
+
+	"universe.dagger.io/examples/todoapp/pkg/yarn"
+)
+
+dagger.#Plan & {
+	client: {
+		filesystem: {
+			"./_build": write: contents: actions.build.output
+			"./app/": write: contents:   actions.develop.output
+			"./app": read: {
+				contents: dagger.#FS
+				exclude: [
+					".git",
+					"node_modules",
+					"*.md",
+				]
+			}
+		}
+		env: {
+			NETLIFY_TOKEN: dagger.#Secret
+			APP_NAME:      string
+		}
+	}
+	actions: {
+		// Checkout a fresh copy of the todoapp repo
+		develop: git.#Pull & {
+			remote: "https://github.com/mdn/todo-react"
+			ref:    "master"
+		}
+
+		yarn.#App & {
+			source: client.filesystem."./app".read.contents
+			name:   "todoapp"
+		}
```

Comment:
Does this extraction make things clearer for someone that is only getting started with Dagger?

I am seeing less cohesion, more cognitive work and something which is just more difficult to understand.

Arguably, this might make sense as a next iteration, post-getting started.

Reply:
Yes, I find it much clearer. Less lines to read, less repetition, and less coupling to `client.filesystem` which is a negative point of our DX at the moment.